plugin/core/sessions.py

```diff
@@ -731,6 +732,7 @@ def m_workspace_configuration(self, params: Dict[str, Any], request_id: Any) ->
         for requested_item in requested_items:
             configuration = self.config.settings.get(requested_item.get('section') or None)
             if self._plugin:
+                configuration = deepcopy(configuration)
```

Comment:
Perhaps it’s time to add a dedicated copy method to ClientConfig because we’re falling into this trap in a few more places I think. Please add a copy method.

Reply:
`self.config.settings` is a DottedDict but `configuration` is a plain dict here. Are you saying we should copy the whole dotted dict and then get the `section` from that? That will be slightly less efficient.

Reply:
On the other hand `section` is typically an empty string...

Reply:
Oh, I meant ClientConfig :D but it's good to have a copy method for DottedDict too.

Reply:
You're asking to have a way to copy a ClientConfig, including deepcopy'ing all its properties? Why would that be useful?

Reply:
Well, each window should have its own set of ClientConfigs because of project overrides.